sys/net/gnrc/netif/lorawan/gnrc_netif_lorawan.c

```diff
@@ -351,11 +351,12 @@ static int _set(gnrc_netif_t *netif, const gnrc_netapi_opt_t *opt)
     switch (opt->opt) {
         case NETOPT_LORAWAN_DR:
             assert(opt->data_len == sizeof(uint8_t));
-            if (*((uint8_t *)opt->data) & 0xF0) {
+            if (!gnrc_lorawan_validate_dr(*((uint8_t *)opt->data))) {
                 DEBUG("gnrc_netif: Invalid datarate\n")
-                gnrc_netif_release(netif);
-                return -EINVAL;
+                res = -EINVAL;
             }
+            else
+            {
             netif->lorawan.datarate = *((uint8_t *)opt->data);
```

Comment:
please fix identation (and add missing `{`)

Reply:
Yes, fixup added :)